File: php/testcode/benchmark_test_00924.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00924(BenchmarkRequest $req): BenchmarkResponse {
    $name  = htmlspecialchars($req->param('name'));
    $email = htmlspecialchars($req->param('email'));
    $out = compact('name', 'email');
    return BenchmarkResponse::json($out);
}

File: rust/testcode/benchmark_test_00924.rs
fn mongo_find(collection: &str, filter: &str) -> String {
    format!("Query on {} with filter: {}", collection, filter)
}

pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let mut filters: Vec<String> = Vec::new();
    filters.push("{\"active\":true}".to_string());
    filters.push(format!("{{\"user\": \"{}\"}}", req.param("u")));
    filters.pop();
    let filter = filters.get(0).map(String::as_str).unwrap_or("{\"active\":true}");
    let result = mongo_find("sessions", filter);
    super::shared::BenchmarkResponse::ok(&result)
}
